components/builder-jobsrv/src/server/worker_manager.rs

```diff
@@ -235,37 +249,118 @@ impl WorkerMgr {
         Ok(())
     }
 
-    fn expire_workers(&mut self) {
+    fn expire_workers(&mut self) -> Result<()> {
         let now = Instant::now();
         loop {
-            if let Some((_, expiry)) = self.workers.front() {
+            if let Some((_, expiry)) = self.ready_workers.front() {
                 if expiry >= &now {
                     break;
                 }
             } else {
                 break;
             }
-            let worker = self.workers.pop_front();
-            debug!("expiring worker due to inactivity, worker={:?}", worker);
+
+            let (worker, _) = self.ready_workers.pop_front().unwrap();
```

Comment:
No, we check the existence of an available entry in the if statement above and break if it is empty.